lib/inv_multi_cg_quda.cpp

```diff
@@ -497,49 +497,53 @@ namespace quda {
     param.gflops = gflops;
     param.iter += k;
 
-    if (param.compute_true_res) {
+    if (param.compute_true_res){
       // only allocate temporaries if necessary
       csParam.setPrecision(param.precision);
       ColorSpinorField *tmp4_p = reliable ? y[0] : tmp1.Precision() == x[0]->Precision() ? &tmp1 : ColorSpinorField::Create(csParam);
       ColorSpinorField *tmp5_p = mat.isStaggered() ? tmp4_p :
-	reliable ? y[1] : (tmp2.Precision() == x[0]->Precision() && &tmp1 != tmp2_p) ? tmp2_p : ColorSpinorField::Create(csParam);
-
-      for(int i=0; i < num_offset; i++) {
-        if(param.precision == param.precision_sloppy || i ==0){
-	mat(*r, *x[i], *tmp4_p, *tmp5_p);
-	if (r->Nspin()==4) {
-	  blas::axpy(offset[i], *x[i], *r); // Offset it.
-	} else if (i!=0) {
-	  blas::axpy(offset[i]-offset[0], *x[i], *r); // Offset it.
-	}
-	double true_res = blas::xmyNorm(b, *r);
-	param.true_res_offset[i] = sqrt(true_res/b2);
-	param.iter_res_offset[i] = sqrt(r2[i]/b2);
-	param.true_res_hq_offset[i] = sqrt(blas::HeavyQuarkResidualNorm(*x[i], *r).z);
-      } else
-      {
-        param.iter_res_offset[i] = sqrt(r2[i]/b2);
-      }
+      reliable ? y[1] : (tmp2.Precision() == x[0]->Precision() && &tmp1 != tmp2_p) ? tmp2_p : ColorSpinorField::Create(csParam);
+
+      for (int i = 0; i < num_offset; i++) {
+        // only calculate true residual if we do not need to
```

Comment:
I suspect we shouldn't only be computing the true residual if we don't need to 😉